gateway-service/src/main/java/org/zowe/apiml/gateway/metadata/service/RefreshEventListener.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+package org.zowe.apiml.gateway.metadata.service;
+
+import org.springframework.cloud.client.discovery.event.HeartbeatEvent;
+import org.springframework.context.ApplicationEvent;
+import org.springframework.context.ApplicationListener;
+
+public abstract class RefreshEventListener implements ApplicationListener<ApplicationEvent> {
+
+    @Override
+    public void onApplicationEvent(ApplicationEvent event) {
+        if (isRefreshEvent(event)) {
+            refresh();
+        }
+    }
+
+    public abstract void refresh();
+
+
+    boolean isRefreshEvent(ApplicationEvent event) {
+        return event instanceof HeartbeatEvent;
```

Comment:
I think you are missing several events, in which the routes do get refreshed. 
Compare with cases in
org.springframework.cloud.netflix.zuul.ZuulServerAutoConfiguration.ZuulRefreshListener#onApplicationEvent

Reply:
added events same as in ZullServerAutoConfiguration